palette/src/oklch.rs

```diff
@@ -0,0 +1,727 @@
+{
+    type X = Oklch<T>;
+
+    fn new<B1, B2>(low_b: B1, high_b: B2) -> Self
+    where
+        B1: SampleBorrow<Self::X> + Sized,
+        B2: SampleBorrow<Self::X> + Sized,
+    {
+        let low = *low_b.borrow();
+        let high = *high_b.borrow();
+
+        UniformOklch {
+            l: Uniform::new::<_, T>(low.l, high.l),
+            chroma: Uniform::new::<_, T>(low.chroma * low.chroma, high.chroma * high.chroma),
+            hue: crate::hues::UniformOklabHue::new(low.hue, high.hue),
+        }
+    }
+
+    fn new_inclusive<B1, B2>(low_b: B1, high_b: B2) -> Self
+    where
+        B1: SampleBorrow<Self::X> + Sized,
+        B2: SampleBorrow<Self::X> + Sized,
+    {
+        let low = *low_b.borrow();
+        let high = *high_b.borrow();
+
+        UniformOklch {
+            l: Uniform::new_inclusive::<_, T>(low.l, high.l),
+            chroma: Uniform::new_inclusive::<_, T>(
+                low.chroma * low.chroma,
+                high.chroma * high.chroma,
+            ),
+            hue: crate::hues::UniformOklabHue::new_inclusive(low.hue, high.hue),
+        }
+    }
+
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> Oklch<T> {
+        Oklch {
+            l: self.l.sample(rng),
+            chroma: crate::Float::sqrt(self.chroma.sample(rng)),
+            hue: self.hue.sample(rng),
+        }
+    }
+}
+
+#[cfg(feature = "bytemuck")]
+unsafe impl<T> bytemuck::Zeroable for Oklch<T> where T: FloatComponent + bytemuck::Zeroable {}
+
+#[cfg(feature = "bytemuck")]
+unsafe impl<T> bytemuck::Pod for Oklch<T> where T: FloatComponent + bytemuck::Pod {}
+
+#[cfg(test)]
+mod test {
+    use crate::Oklch;
+
+    #[test]
+    fn ranges() {
+        assert_ranges! {
+            Oklch< f64>;
+            clamped {
+                l: 0.0 => 1.0,
+                chroma: 0.0 => 1.0
+            }
+            clamped_min {}
+            unclamped {
+                hue: -360.0 => 360.0
+            }
+        }
+    }
+
+    #[test]
+    fn check_min_max_components() {
+        assert_relative_eq!(Oklch::<f32>::min_l(), 0.0);
+        assert_relative_eq!(Oklch::<f32>::max_l(), 1.0);
+        assert_relative_eq!(Oklch::<f32>::min_chroma(), 0.0);
+        assert_relative_eq!(Oklch::<f32>::max_chroma(), 1.0);
+    }
+
+    #[cfg(feature = "serializing")]
+    #[test]
+    fn serialize() {
+        let serialized = ::serde_json::to_string(&Oklch::new(0.3, 0.8, 0.1)).unwrap();
+
+        assert_eq!(serialized, r#"{"l":0.3,"chroma":0.8,"hue":0.1}"#);
+    }
+
+    #[cfg(feature = "serializing")]
+    #[test]
+    fn deserialize() {
+        let deserialized: Oklch =
+            ::serde_json::from_str(r#"{"l":0.3,"chroma":0.8,"hue":0.1}"#).unwrap();
+
+        assert_eq!(deserialized, Oklch::new(0.3, 0.8, 0.1));
+    }
+
+    #[cfg(feature = "random")]
+    test_uniform_distribution! {
+        Oklch<f32> as crate::Oklab {
+            l: (0.0, 1.0),
+            a: (-1.0, 1.0),
+            b: (-1.0, 1.0),
+        },
```

Comment:
I think I know what's going on with the random numbers. I remember having the same issue before. You need to make the limits for `a` and `b` smaller, to `+/-0.7` (`cos 45°`) to make it look for a square that will fit inside the `OkLch` circle. Otherwise it will not generate any points in the corners and the average density becomes uneven.